### PR TITLE
New Select component in UI package

### PR DESCRIPTION
**Pull request by ccschmitz**
## Summary

Creates a new `Select` component in the UI package and sets it up in a few places in the app. There is more work to do to swap out other instances of antd selects which will be handled in future PRs.

Check out the Storybook stories in Reflame or locally and play around with it to see how it works. I think the stories show off most of the features, but here's a list of what's currently supported as well:

* Basic single selects
* Multi select
* Filterable selects
* Controlled selects (value managed externally)
* "tags" display mode
* Custom value rendering (used under the hood for tags)
* A `loading` prop if data is being fetched for options when initially rendered
* A `clearable` prop for selects that you can set to an empty value

## How did you test this change?

I wrote a number of Storybook stories that I clicked tested thoroughly. I also used the new in our frontend a few places to ensure it worked as expected.

### CODE REVIEW

File: packages/ui/src/components/Button/Button.stories.tsx
Comment (ccschmitz) [inline packages/ui/src/components/Button/Button.stories.tsx:45]: Fixed these stories which weren't rendering icons correctly

File: packages/ui/package.json
Comment (ccschmitz) [inline packages/ui/package.json:114]: Some of the Storybook config changes came from running the auto upgrade task from the Storybook CLI.

File: frontend/src/pages/Alerts/LogAlert/LogAlertPage.tsx
Comment (SpennyNDaJets) [inline frontend/src/pages/Alerts/LogAlert/LogAlertPage.tsx:1]: Very nitty: but when you change the Alert frequency, the width of the select changes here

File: frontend/src/pages/WorkspaceTeam/components/AutoJoinForm.tsx
Comment (SpennyNDaJets) [inline frontend/src/pages/WorkspaceTeam/components/AutoJoinForm.tsx:1]: Seeing an error that may be related when trying to access:
- https://preview.highlight.io/w/1/settings
- https://preview.highlight.io/w/1/team

```
TypeError: Cannot read properties of undefined (reading 'length')
```

**Review by SpennyNDaJets — CHANGES_REQUESTED** [review]
Looking great! Just left a couple comments from click test:

1. On `ErrorAlertPage`, not able to select an option
Screenshot 2024-08-07 at 4 10 37 PM

2. Also saw it was used on the SessionAlertForm, but wondering if `configuration.canControlThreshold ` should just be cleaned up?

File: frontend/src/pages/WorkspaceTeam/components/AutoJoinForm.tsx
Comment (ccschmitz) [reply frontend/src/pages/WorkspaceTeam/components/AutoJoinForm.tsx:1]: Thanks for catching this! Just fixed that up. It was an issue with auto join form.

File: frontend/src/pages/Alerts/LogAlert/LogAlertPage.tsx
Comment (ccschmitz) [reply frontend/src/pages/Alerts/LogAlert/LogAlertPage.tsx:1]: Fixed by setting `span="6"` on each wrapping `Column` - good catch!

**Comment by ccschmitz** [comment]
> ... wondering if `configuration.canControlThreshold ` should just be cleaned up?

Maybe so, but I might hold off on that in this PR. Will defer to you on that one with the work you're already doing on alerts 😸 

I think everything else is cleaned up, so this is ready for another look! Thanks for the review!

File: packages/ui/src/components/Form/styles.css.ts
Comment (ccschmitz) [inline packages/ui/src/components/Form/styles.css.ts:86]: Noticed the height of form inputs was slightly off from the mocks. All form inputs were supposed to be 28px tall, which this enforces.

**Review by SpennyNDaJets — APPROVED** [review]
Click test and code look good to me!

**Comment by ccschmitz** [comment]
Thanks for the review on this, @SpennyNDaJets! Apologies that this got so large...

**Pull request merged**
